plugins/automation/spec/system/smoke_test_spec.rb

```diff
@@ -33,6 +33,22 @@
       select_kit.select_row_by_value("post_created_edited")
 
       expect(find(".field input[name=test]").value).to eq("test-default-value")
+
+      automation = Fabricate(:automation, name: "automation-test")
+
+      visit("/admin/plugins/automation")
+      # find the row of "automation test" then click on trash icon
+      find(".automations__name", text: "automation-test")
+        .find(:xpath, "..")
+        .find(".automations__delete")
+        .click
+
+      find(".dialog-footer .btn-danger").click
+
+      expect(page).not_to have_css(".automations__name", text: "automation-test")
+
+      automation = DiscourseAutomation::Automation.find_by(id: automation.id)
+      expect(automation).to be_nil
```

Comment:
This is a common source of test flakiness. We clicked a button to trigger an async operation but is asserting straight away for a state in the database. From a system test perspective, it is also not recommended to test for state in the database as a system test is supposed to be testing how a user interacts with the application from their perspective where the application is essentially a black box for them. If we must test that the record has been deleted, the easiest way is to reload the page and check that the record is no longer loaded. If we must check for state in the database, we need to retry a couple of times to account for the fact that the operation is an asynchronous one.
```suggestion
      try_until_success do
        expect(DiscourseAutomation::Automation.exists?(id: automation.id)).to be(false)
      end  
```